.github/workflows/ci.yml

```diff
@@ -29,7 +29,34 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  build:
+  diff:
+    timeout-minutes: 2
+    runs-on: ubuntu-latest
+    name: "Diff files"
+    outputs:
+      runTest: ${{ steps.skip-tests-files.outputs.only_changed != 'true' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # Assume PRs are less than 50 commits
+          fetch-depth: 50
+
+      - name: Get changed files
+        id: skip-tests-files
+        uses: tj-actions/changed-files@v35
```

Comment:
is it possible to use a specific hash here? by changing the tag someone with access to that actions repo could do nefarious things

Reply:
For this build I think it's safe, the action doesn't have more access than any existing contributor that could just update this file. There is no token present in this job.

I can still add the git hash if you prefer

Reply:
it does have access to secrets when run on push event. these could be extracted. so yes. using a hash would be better. on the other hand later on there is a curl bash, if that's ok, so would be using a tag here. 

imho neither is, so I'd prefer hash ref and auditing what that action does, as well as removing the curl bash ( in a different PR)

Reply:
Secrets are not available [without inputs](https://docs.github.com/en/actions/security-guides/encrypted-secrets#accessing-your-secrets), the bash script/action here have the same access than hundreds of dependencies of eslint & playwright, so I think we should treat everything in the CI env as public instead of auditing every dev deps of Vite core.

Reply:
see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#restricting-permissions-for-tokens

as far as I understand this, you have to modify permissions for GITHUB_TOKEN for your statement to hold true.
And even if it is, they still recommend to only use tags for third party actions if you trust the author in that guide.

I do agree about dependencies btw and we should look into whitelisting postinstall scritpts of trusted deps only (esbuild comes to mind). This is possible with pnpm but again subject to a different PR.

Reply:
The only tokens in the Vite repo are the `NPM_TOKEN` and the two tokens for vite-ecosystem-ci (`ECOSYSTEM_CI_GITHUB_APP_ID`, `ECOSYSTEM_CI_GITHUB_APP_PRIVATE_KEY`). Maybe I'm missing some context of what I should be looking for here.

Reply:
Sorry I didn't read the docs enough. For job level permissions, this is configurable directly in the yaml:
https://github.com/vitejs/vite/pull/11548/commits/0590c8eec94f541ff46e0a3b0428fbaab60f26ca

Remove all permissions seems to work!